### ISSUE TITLE
dedupe delete job: ValueError('Cannot use None as a query value',)

### ISSUE THREAD

**Issue by valentijnscholten** [DefectDojo/django-DefectDojo#3368]
Seen this in some logs submitted in other issues, let's investigate.
```
Received task: dojo.tasks.async_dupe_delete[b9ddfdba-d0c1-452e-8b32-24dc4acde1eb]
delete excess duplicates
delete excess duplicates
Task dojo.tasks.async_dupe_delete[b9ddfdba-d0c1-452e-8b32-24dc4acde1eb] raised unexpected: ValueError('Cannot use None as a query value',)
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/celery/app/trace.py", line 412, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/celery/app/trace.py", line 704, in __protected_call__
    return self.run(*args, **kwargs)
  File "/app/dojo/tasks.py", line 333, in async_dupe_delete
```

**Comment by valentijnscholten** [comment]
seen in #3332 but that line number doesn't exist in 1.10.0.
Line in 1.9.3: https://github.com/DefectDojo/django-DefectDojo/blob/a1d489069aa0347d1a05bce0fe24a92226d70a96/dojo/tasks.py#L333

**Comment by Sabartius** [comment]
Having the error in 1.10.4:
https://github.com/DefectDojo/django-DefectDojo/blob/09cf49203aaf4f25533f601da9e47ab3791450f4/dojo/tasks.py#L234

```
[10/Dec/2020 15:42:06] ERROR [celery.app.trace:257] Task dojo.tasks.async_dupe_delete[32d22b2f-9fd2-4c5d-a1d2-193806289f8a] raised unexpected: ValueError('Cannot use None as a query value',)
  | Traceback (most recent call last):
  | File "/usr/local/lib/python3.6/site-packages/celery/app/trace.py", line 412, in trace_task
  | R = retval = fun(*args, **kwargs)
  | File "/usr/local/lib/python3.6/site-packages/celery/app/trace.py", line 704, in __protected_call__
  | return self.run(*args, **kwargs)
  | File "/app/dojo/tasks.py", line 234, in async_dupe_delete
  | .filter(num_dupes__gt=dupe_max)
  | File "/usr/local/lib/python3.6/site-packages/django/db/models/query.py", line 892, in filter
  | return self._filter_or_exclude(False, *args, **kwargs)
  | File "/usr/local/lib/python3.6/site-packages/django/db/models/query.py", line 910, in _filter_or_exclude
  | clone.query.add_q(Q(*args, **kwargs))
  | File "/usr/local/lib/python3.6/site-packages/django/db/models/sql/query.py", line 1290, in add_q
  | clause, _ = self._add_q(q_object, self.used_aliases)
  | File "/usr/local/lib/python3.6/site-packages/django/db/models/sql/query.py", line 1318, in _add_q
  | split_subq=split_subq, simple_col=simple_col,
  | File "/usr/local/lib/python3.6/site-packages/django/db/models/sql/query.py", line 1207, in build_filter
  | condition = self.build_lookup(lookups, reffed_expression, value)
  | File "/usr/local/lib/python3.6/site-packages/django/db/models/sql/query.py", line 1121, in build_lookup
  | raise ValueError("Cannot use None as a query value")
  | ValueError: Cannot use None as a query value


```

**Comment by Sabartius** [comment]
This Error occurs when deduplicating Dependency-Track Findings (FPE), which we configured to HASH_CODE_ALGO and using following params `'Dependency Track Finding Packaging Format (FPF) Export': ['cwe', 'severity', 'file_path'],`
This worked fine with 1.6.5 but after Upgrading to 1.10.4 instead of duplicates we get the error metioned above. Any Ideas @valentijnscholten ?

**Comment by michaelgibson** [comment]
Was getting this as well
Setting "Max Duplicates" in System Settings to something seems to stop celery from throwing that error constantly.
May want to have a default for this?
![image](https://user-images.githubusercontent.com/2904692/102287027-cb61e700-3ef6-11eb-8e43-d678c9645bbb.png)

**Comment by valentijnscholten** [comment]
Aha. When first looking at the traceback I thought this would only occur if there were no duplicates found. But it's when no maximum number of duplicates is set. Good to have an easy work arkound, we'll see about a default.

**Comment by valentijnscholten** [comment]
"adding a quick default" turned into "refactoring the whole dedupe job", see PR.

**Comment by valentijnscholten** [comment]
Fixed in dev / 1.11.0
